pinot-common/src/main/java/com/linkedin/pinot/common/config/TableConfig.java

```diff
@@ -184,22 +198,44 @@ public void setQuotaConfig(@Nullable QuotaConfig quotaConfig) {
     _quotaConfig = quotaConfig;
   }
 
+  @Nullable
+  public TableTaskConfig getTaskConfig() {
+    return _taskConfig;
+  }
+
+  public void setTaskConfig(@Nullable TableTaskConfig taskConfig) {
+    _taskConfig = taskConfig;
+  }
+
   @Nonnull
-  public JSONObject toJSON()
+  public static JSONObject getJSONConfig(@Nonnull String tableName, @Nonnull TableType tableType,
+      @Nonnull SegmentsValidationAndRetentionConfig validationConfig, @Nonnull TenantConfig tenantConfig,
+      @Nonnull IndexingConfig indexingConfig, @Nonnull TableCustomConfig customConfig,
+      @Nullable QuotaConfig quotaConfig, @Nullable TableTaskConfig taskConfig)
       throws IOException, JSONException {
     JSONObject jsonConfig = new JSONObject();
-    jsonConfig.put(TABLE_NAME_KEY, _tableName);
-    jsonConfig.put(TABLE_TYPE_KEY, _tableType.toString().toLowerCase());
-    jsonConfig.put(VALIDATION_CONFIG_KEY, new JSONObject(OBJECT_MAPPER.writeValueAsString(_validationConfig)));
-    jsonConfig.put(TENANT_CONFIG_KEY, new JSONObject(OBJECT_MAPPER.writeValueAsString(_tenantConfig)));
-    jsonConfig.put(INDEXING_CONFIG_KEY, new JSONObject(OBJECT_MAPPER.writeValueAsString(_indexingConfig)));
-    jsonConfig.put(CUSTOM_CONFIG_KEY, new JSONObject(OBJECT_MAPPER.writeValueAsString(_customConfig)));
-    if (_quotaConfig != null) {
-      jsonConfig.put(QUOTA_CONFIG_KEY, new JSONObject(OBJECT_MAPPER.writeValueAsString(_quotaConfig)));
+    jsonConfig.put(TABLE_NAME_KEY, tableName);
```

Comment:
We should avoid creating a JSONObject from strings equivalent of individual fields. It would be better if we can directly use object mapper to generate json object from the TableConfig object. Not a blocker for this PR, because that is likely a bigger change.